src/domain/collection/curatedEventList/__tests__/CuratedEventList.test.tsx

```diff
@@ -21,19 +23,62 @@ const collection = fakeCollection({
   curatedEvents,
 }) as CollectionFieldsFragment;
 
-const getMocks = (events: EventFieldsFragment[]) => [
+const getMocks = (events: EventFieldsFragment[], ids = eventIds) => [
   {
     request: {
       query: EventsByIdsDocument,
       variables: {
-        ids: eventIds,
+        ids,
         include: ['location'],
       },
     },
     result: { data: { eventsByIds: events } },
   },
 ];
 
+// Creates array of mocks to match pagination queries
+const getMocksForPagination = (eventsCount = 35) => {
+  const eventIds = range(eventsCount).map((i) => (i + 1).toString());
+  const eventNames = eventIds.map((id) => `Event ${id}`);
+  const curatedEvents = eventIds.map(
+    (id) => `http://localhost:3000/fi/event/${id}`
+  );
+
+  const chunkedEventIds = chunk(eventIds, PAGE_SIZE);
+  const chunkedEvents = chunk(eventNames, PAGE_SIZE).map(
+    (splittedEventNames, chunkIndex) => {
+      return fakeEvents(
+        splittedEventNames.length,
+        splittedEventNames.map((event, index) => ({
+          endTime: '2020-12-12',
+          startTime: '2020-12-12',
+          id: chunkedEventIds[chunkIndex][index],
+          name: { fi: event },
+        }))
+      );
+    }
+  );
+
+  const mocks = chunkedEventIds.map((eventIds, index) => {
+    return getMocks(
+      chunkedEvents[index].data as EventFieldsFragment[],
+      eventIds
+    )[0];
+  });
+
+  const collection = fakeCollection({
+    curatedEvents,
+  }) as CollectionFieldsFragment;
+
+  return { mocks, collection, eventNames };
+};
+
+const waitForRequestToComplete = async () => {
```

Comment:
at some point this could be moved to some util method as it is used like 100s of different tests :)

Reply:
yeah at least for the loading spinner